api.ts

```diff
@@ -1852,6 +1909,30 @@ export function makeDynamicSyncTable<
   L extends string,
   ParamDefsT extends ParamDefs,
   SchemaT extends ObjectSchemaDefinition<K, L>,
+>(params: {
+  name: string;
+  description?: string;
+  getName: MetadataFormulaDef;
+  getSchema: MetadataFormulaDef;
+  identityName: string;
+  formula: SyncFormulaDef<K, L, ParamDefsT, any>;
+  getDisplayUrl: MetadataFormulaDef;
+  listDynamicUrls?: MetadataFormulaDef;
+  entityName?: string;
+  connectionRequirement?: ConnectionRequirement;
+  defaultAddDynamicColumns?: boolean;
+  placeholderSchema?: SchemaT;
+}): DynamicSyncTableDef<K, L, ParamDefsT, any>;
+
+/**
+ * Includes the unreleased propertyAutocomplete parameter.
+ * @hidden
```

Comment:
Used function overloading + @hidden here to avoid `propertyAutocomplete` appearing in documentation

Reply:
Seems like it's possible to extract an `export`ed `MakeDynamicSyncTableParams` type where one of its fields is `@hidden`. Seems like extracting an exported type for the parameters. I poked around for a few minutes, but I think this temporary code duplication is probably the simplest and least disruptive option I tried